AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Service/Storage/List/StorageListingErrorTransformer.swift

```diff
@@ -0,0 +1,107 @@
+    func transform(sdkError: SdkError<ListObjectsV2OutputError>) -> StorageError {
+        switch sdkError {
+        case .unknown:
+            return StorageError(error: sdkError)
+        case .client(let clientError, let response):
+            guard let response = response else {
+                return transform(clientError: clientError)
+            }
+            if let result = transform(statusCode: response.statusCode.rawValue, error: sdkError) {
+                return result
+            }
+        case .service(let error, let response):
+            if let result = transform(statusCode: response.statusCode.rawValue, error: error) {
+                return result
+            }
+        }
+        return StorageError(error: sdkError)
+    }
+    
+    /// - Tag: StorageListingErrorTransformer.transform_listClientError
+    private func transform(clientError: ClientError) -> StorageError {
+        switch clientError {
+        case .authError(let message):
+            return StorageError.authError(clientError.localizedDescription, message)
+        case .crtError,
+                .networkError,
+                .pathCreationFailed,
+                .serializationFailed,
+                .deserializationFailed,
+                .dataNotFound,
+                .unknownError:
+            return StorageError(error: clientError)
+        case .retryError(let underlyingError):
+            if let serviceError = underlyingError as? SdkError<ListObjectsV2OutputError> {
+                return transform(sdkError: serviceError)
+            }
+            return StorageError(error: clientError)
+        }
+    }
+    
+    /// - Tag: StorageListingErrorTransformer.transform_statusCode
+    private func transform(statusCode: Int, error: Error) -> StorageError? {
+        if statusCode / 100 == 3, statusCode != 304 {
+            // 300 range: Redirection
+            return .httpStatusError(statusCode, "Redirection error", error)
+        } else if statusCode / 100 == 4 {
+            // 400 range: Client Error
+            let description = Self.errorDescription(forStatusCode: statusCode)
+            if [401, 403].contains(statusCode) {
+                return .accessDenied(description,
+                                     "Make sure the user has access to the key before trying to download/upload it.",
+                                     error)
+            } else if 404 == statusCode {
+                return .keyNotFound(key ?? "",
+                                    description,
+                                    "Make sure the key exists before trying to download it.",
+                                    error)
+            } else {
+                return .httpStatusError(statusCode, "Client error", error)
+            }
+        } else if statusCode / 100 == 5 {
+            // 500 range: Server Error
+            return .httpStatusError(statusCode, "Server error", error)
+        }
+        return nil
```

Comment:
non-blocking readability suggestion:
```suggestion
    switch statusCode {
    case 300...303, 305...399:
        // 300 range: Redirection
        return .httpStatusError(statusCode, "Redirection error", error)
    case 401, 403:
        // 400 range: Client Error
        return .accessDenied(
            description,
            "Make sure the user has access to the key before trying to download/upload it.",
            error
        )
    case 404:
        // 400 range: Client Error
        return .keyNotFound(
            key ?? "",
            description,
            "Make sure the key exists before trying to download it.",
            error
        )
    case 400...499:
        // 400 range: Client Error
        return .httpStatusError(statusCode, "Client error", error)
    case 500...599:
        // 500 range: Server Error
        return .httpStatusError(statusCode, "Server error", error)
    default:
        return nil
    }
```

Reply:
Yes, sounds great! I have to admit I copied this from `StorageTransferResponse.swift`